src/charts/bar/Bar.js

```diff
@@ -158,36 +158,40 @@ class Bar extends Component {
         shouldShowLoadingState: false,
     }
 
-    constructor(props) {
-        super(props);
+    componentDidMount() {
+        if (this.props.data !== null) {
+            this._createChart();
+        }
+    }
 
-        // We want to make this throw an error if no data is provided
-        if (!props.data) {
-            throw new Error('Data is required!');
+    componentDidUpdate(prevProps) {
+        if (prevProps.data === null && this.props.data) {
```

Comment:
And this will be more about the shouldShowLoadingState, right?

Reply:
I don't think we need it, but it could be a performance optimization.